http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/MultiplexedChannelRecord.java

```diff
@@ -111,6 +112,7 @@ void acquireClaimedStream(Promise<Channel> promise) {
                 Http2StreamChannel channel = future.getNow();
                 channel.pipeline().addLast(UnusedChannelExceptionHandler.getInstance());
                 channel.attr(ChannelAttributeKey.HTTP2_FRAME_STREAM).set(channel.stream());
+                channel.attr(ChannelAttributeKey.CHANNEL_DIAGNOSTICS).set(new ChannelDiagnostics());
```

Comment:
Is it intentional to not initialize channel creation time here, like we do in `ChannelPipelineInitializer`?

Reply:
No, and I should probably initialize it at creation time to prevent making that mistake in the future. Good catch.